pangeo_forge/recipe.py

```diff
@@ -197,37 +215,44 @@ def _prepare_target():
                     pass
 
             except (FileNotFoundError, IOError, zarr.errors.GroupNotFoundError):
-
-                init_dsets = []
+                logger.info("Creating a new dataset in target")
+                # need to rewrite this as an append loop
                 for chunk_key in self._init_chunks:
-                    chunk_ds = self.open_chunk(
-                        chunk_key
-                    ).chunk()  # make sure data are not in memory
-                    init_dsets.append(chunk_ds)
-                # TODO: create csutomizable option for this step
-                # How to combine attrs is particularly important. It seems like
-                # xarray is missing a "minimal" option to only keep the attrs
-                # that are the same among all input variables.
-                ds = xr.merge(
-                    init_dsets, compat="identical", join="exact", combine_attrs="override"
-                )
-
-                # make sure the concat dim has a valid fill_value to avoid
-                # overruns when writing chunk
-                # this is problematic because this value will always come out as NaN
-                # when we open up the target
-                # ds[self.sequence_dim].encoding = {"_FillValue": -1}
-                # actually not necessary if we use decode_times=False
-
-                if self.target_chunks:
-                    # target_chunks has undergone some validation in __post_init__
-                    ds = ds.chunk(self.target_chunks)
-
-                self.initialize_target(ds)
+                    with self.open_chunk(chunk_key) as ds:
+                        # need to have the data in memory to avoid weird chunk problems
+                        ds.load()
```

Comment:
This is really not ideal. Can we avoid it somehow and keep the data loading lazy?